# 상권 - 서비스 업종 추정매출, 폐업률

In [1]:
import pandas as pd

## 1. 데이터 불러오기

### (1) 업종별 점포 데이터 

In [3]:
# 2020년 = service_info_2020, 
# 2019년 = service_info_2019, 
# .... 
# 2015년 = service_info_2015

for i in range(2015, 2021):
    globals()['service_info_{}'.format(i)] = pd.read_csv(path + str(i) + '.csv', encoding = 'cp949')

In [4]:
# 확인

service_info_2020.head()

기준_년_코드  기준_분기_코드 상권_구분_코드 상권_구분_코드_명    상권_코드      상권_코드_명 서비스_업종_코드  \
0     2020         2        U       관광특구  1001496  강남 마이스 관광특구  CS300043   
1     2020         2        U       관광특구  1001496  강남 마이스 관광특구  CS300042   
2     2020         2        U       관광특구  1001496  강남 마이스 관광특구  CS300041   
3     2020         2        U       관광특구  1001496  강남 마이스 관광특구  CS300040   
4     2020         2        U       관광특구  1001496  강남 마이스 관광특구  CS300038   

  서비스_업종_코드_명  점포_수  유사_업종_점포_수  개업_율  개업_점포_수  폐업_률  폐업_점포_수  프랜차이즈_점포_수  
0      전자상거래업     8           8     0        0     0        0           0  
1         주유소     1           1     0        0     0        0           0  
2         예술품     9           9     0        0     0        0           0  
3    재생용품 판매점     0           0     0        0     0        0           0  
4       자동차부품    10          10     0        0     0        0           0

In [122]:
food = ['한식음식점', '중식음식점', '일식음식점', '양식음식점', '제과점', '패스트푸드점', '치킨전문점', '분식전문점', '호프-간이주점',
       '커피-음료']

service = ['일반교습학원', '외국어학원', '예술학원', '컴퓨터학원', '스포츠 강습', '일반의원', '치과의원', '한의원', '동물병원',
          '변호사사무소', '변리사사무소', '법무사사무소', '기타법무서비스', '회계사사무소', '세무사사무소', '당구장',
          '골프연습장', '볼링장', 'PC방', '전자게임장', '기타오락장', '복권방', '통신기기수리', '스포츠클럽', '자동차수리', '자동차미용',
          '모터사이클수리', '미용실', '네일숍', '피부관리실', '세탁소', '가전제품수리', '부동산중개업', '여관', '게스트하우스', '고시원',
          '노래방', '독서실', 'DVD방', '녹음실', '사진관', '통번역서비스', '건축물청소', '여행사', '비디오/서적임대', '의류임대', 
          '가정용품임대']

retail = ['슈퍼마켓', '편의점', '컴퓨터및주변장치판매', '핸드폰', '주류도매', '미곡판매', '육류판매', '수산물판매', '청과상',
          '반찬가게', '일반의류', '한복점', '유아의류', '신발', '가방', '안경', '시계및귀금속', '의약품', '의료기기', '서적', 
         '문구', '화장품', '미용재료', '운동/경기용품', '자전거 및 기타운송장비', '완구', '섬유제품', '화초', '애완동물', '중고가구', 
         '가구', '가전제품', '철물점', '악기', '인테리어', '조명용품', '중고차판매', '자동차부품', '모터사이클및부품', 
          '재생용품 판매점', '예술품', '주유소', '전자상거래업']

In [130]:
food_code = []
service_code = []
retail_code = []

food_name = []
service_name = []
retail_name = []

In [131]:
for i in service_info_2020.index:
    r = service_info_2020.loc[i,'서비스_업종_코드_명']
    if r in food:
        food_code.append(service_info_2020.loc[i,'서비스_업종_코드'])
        food_name.append(service_info_2020.loc[i,'서비스_업종_코드_명'])
    if r in service:
        service_code.append(service_info_2020.loc[i,'서비스_업종_코드'])
        service_name.append(service_info_2020.loc[i,'서비스_업종_코드_명'])
    if r in retail:
        retail_code.append(service_info_2020.loc[i,'서비스_업종_코드'])
        retail_name.append(service_info_2020.loc[i,'서비스_업종_코드_명'])

In [135]:
df1 = pd.DataFrame({'code' : food_code, 'name' : food_name})

In [137]:
df1['code'].unique()

array(['CS100010', 'CS100009', 'CS100008', 'CS100007', 'CS100006',
       'CS100005', 'CS100004', 'CS100003', 'CS100002', 'CS100001'],
      dtype=object)

In [138]:
df1['name'].unique()

array(['커피-음료', '호프-간이주점', '분식전문점', '치킨전문점', '패스트푸드점', '제과점', '양식음식점',
       '일식음식점', '중식음식점', '한식음식점'], dtype=object)

In [123]:
print("food : {}\nservice : {}\nretail : {}".format(len(food), len(service), len(retail)))

food : 10
service : 47
retail : 43


In [124]:
food_df = pd.DataFrame()
service_df = pd.DataFrame()
retail_df = pd.DataFrame()

for f in food:
    food_df = food_df.append(service_info_2020[service_info_2020['서비스_업종_코드_명'] == f])
for s in service:
    service_df = service_df.append(service_info_2020[service_info_2020['서비스_업종_코드_명'] == s])
for r in retail:
    retail_df = retail_df.append(service_info_2020[service_info_2020['서비스_업종_코드_명'] == r])

In [125]:
len(food_df['서비스_업종_코드_명'].unique())

10

In [126]:
len(service_df['서비스_업종_코드_명'].unique())

47

In [127]:
len(retail_df['서비스_업종_코드_명'].unique())

43

In [128]:
len(a)

5

In [129]:
service_df['서비스_업종_코드_명'].unique()

array(['일반교습학원', '외국어학원', '예술학원', '컴퓨터학원', '스포츠 강습', '일반의원', '치과의원',
       '한의원', '동물병원', '변호사사무소', '변리사사무소', '법무사사무소', '기타법무서비스', '회계사사무소',
       '세무사사무소', '당구장', '골프연습장', '볼링장', 'PC방', '전자게임장', '기타오락장', '복권방',
       '통신기기수리', '스포츠클럽', '자동차수리', '자동차미용', '모터사이클수리', '미용실', '네일숍',
       '피부관리실', '세탁소', '가전제품수리', '부동산중개업', '여관', '게스트하우스', '고시원', '노래방',
       '독서실', 'DVD방', '녹음실', '사진관', '통번역서비스', '건축물청소', '여행사', '비디오/서적임대',
       '의류임대', '가정용품임대'], dtype=object)

In [118]:
retail_df['서비스_업종_코드_명'].unique()

array(['슈퍼마켓', '편의점', '컴퓨터및주변장치판매', '핸드폰', '주류도매', '미곡판매', '육류판매',
       '수산물판매', '청과상', '반찬가게', '일반의류', '한복점', '유아의류', '신발', '가방', '안경',
       '시계및귀금속', '의약품', '의료기기', '서적', '문구', '화장품', '미용재료', '운동/경기용품',
       '완구', '섬유제품', '화초', '애완동물', '중고가구', '가구', '가전제품', '철물점', '악기',
       '인테리어', '조명용품', '중고차판매', '자동차부품', '모터사이클및부품', '재생용품 판매점', '예술품',
       '주유소', '전자상거래업'], dtype=object)

In [119]:
service_info_2020.columns

Index(['기준_년_코드', '기준_분기_코드', '상권_구분_코드', '상권_구분_코드_명', '상권_코드', '상권_코드_명',
       '서비스_업종_코드', '서비스_업종_코드_명', '점포_수', '유사_업종_점포_수', '개업_율', '개업_점포_수',
       '폐업_률', '폐업_점포_수', '프랜차이즈_점포_수'],
      dtype='object')

In [132]:
service_info_2019.shape

(365030, 15)

In [134]:
for i in range(2015, 2021):
    dfff = globals()['service_info_{}'.format(i)]
    dlist = []
    for j in dfff.index:
        if (dfff.loc[j,'점포_수'] == 0) and (dfff.loc[j,'유사_업종_점포_수'] == 0) and (dfff.loc[j,'개업_율'] == 0) \
        and (dfff.loc[j,'개업_점포_수'] == 0) and (dfff.loc[j,'폐업_률'] == 0) and (dfff.loc[j,'폐업_점포_수'] == 0) and \
        (dfff.loc[j,'프랜차이즈_점포_수'] == 0):
            dlist.append(j)
    globals()['service_info_{}'.format(i)] = globals()['service_info_{}'.format(i)].drop(dlist).reset_index(drop = True)

In [135]:
service_info_2019.shape

(272050, 15)

In [136]:
for i in range(2015, 2021):
    df = globals()['service_info_{}'.format(i)]
    for j in range(1, 5):
        df_2 = df[df['기준_분기_코드'] == j]
        for k in mlist:
            df_3 = df_2[df_2['상권_코드'] == k]
            globals()['set_{}_{}_{}'.format(i, j, k)] = set(list(df_3['서비스_업종_코드'].unique()))
        if j == 2 and i == 2020:
            break

In [137]:
mlist = [i for i in range(1000001, 1001497)]
len(mlist)

1496

In [138]:
for i in mlist:
    globals()['market_{}'.format(i)] = []

In [139]:
all_service = set(service_info_2020['서비스_업종_코드'].unique())



for service in all_service:
    for i in mlist:
        if (service in globals()['set_2020_2_{}'.format(i)]) and (service in globals()['set_2020_1_{}'.format(i)]) and \
        (service in globals()['set_2019_4_{}'.format(i)]) and (service in globals()['set_2019_3_{}'.format(i)]) and \
        (service in globals()['set_2019_2_{}'.format(i)]) and (service in globals()['set_2019_1_{}'.format(i)]) and \
        (service in globals()['set_2018_4_{}'.format(i)]) and (service in globals()['set_2018_3_{}'.format(i)]) and \
        (service in globals()['set_2018_2_{}'.format(i)]) and (service in globals()['set_2018_1_{}'.format(i)]) and \
        (service in globals()['set_2017_4_{}'.format(i)]) and (service in globals()['set_2017_3_{}'.format(i)]) and \
        (service in globals()['set_2017_2_{}'.format(i)]) and (service in globals()['set_2017_1_{}'.format(i)]) and \
        (service in globals()['set_2016_4_{}'.format(i)]) and (service in globals()['set_2016_3_{}'.format(i)]) and \
        (service in globals()['set_2016_2_{}'.format(i)]) and (service in globals()['set_2016_1_{}'.format(i)]) and \
        (service in globals()['set_2015_4_{}'.format(i)]) and (service in globals()['set_2015_3_{}'.format(i)]) and \
        (service in globals()['set_2015_2_{}'.format(i)]) and (service in globals()['set_2015_1_{}'.format(i)]):
            globals()['market_{}'.format(i)].append(service)
        else:
            continue
                
            

In [140]:
for i in mlist:
    n = len(globals()['market_{}'.format(i)])
    print("{} : {}, {}".format(i ,n, type(globals()['market_{}'.format(i)])))

1000001 : 41, <class 'list'>
1000002 : 27, <class 'list'>
1000003 : 56, <class 'list'>
1000004 : 22, <class 'list'>
1000005 : 26, <class 'list'>
1000006 : 25, <class 'list'>
1000007 : 32, <class 'list'>
1000008 : 20, <class 'list'>
1000009 : 23, <class 'list'>
1000010 : 33, <class 'list'>
1000011 : 40, <class 'list'>
1000012 : 25, <class 'list'>
1000013 : 29, <class 'list'>
1000014 : 19, <class 'list'>
1000015 : 40, <class 'list'>
1000016 : 41, <class 'list'>
1000017 : 39, <class 'list'>
1000018 : 35, <class 'list'>
1000019 : 37, <class 'list'>
1000020 : 39, <class 'list'>
1000021 : 31, <class 'list'>
1000022 : 29, <class 'list'>
1000023 : 25, <class 'list'>
1000024 : 16, <class 'list'>
1000025 : 16, <class 'list'>
1000026 : 27, <class 'list'>
1000027 : 20, <class 'list'>
1000028 : 51, <class 'list'>
1000029 : 18, <class 'list'>
1000030 : 39, <class 'list'>
1000031 : 32, <class 'list'>
1000032 : 54, <class 'list'>
1000033 : 41, <class 'list'>
1000034 : 49, <class 'list'>
1000035 : 31, 

1001407 : 5, <class 'list'>
1001408 : 11, <class 'list'>
1001409 : 39, <class 'list'>
1001410 : 11, <class 'list'>
1001411 : 50, <class 'list'>
1001412 : 36, <class 'list'>
1001413 : 45, <class 'list'>
1001414 : 39, <class 'list'>
1001415 : 36, <class 'list'>
1001416 : 11, <class 'list'>
1001417 : 19, <class 'list'>
1001418 : 41, <class 'list'>
1001419 : 19, <class 'list'>
1001420 : 38, <class 'list'>
1001421 : 13, <class 'list'>
1001422 : 28, <class 'list'>
1001423 : 20, <class 'list'>
1001424 : 40, <class 'list'>
1001425 : 2, <class 'list'>
1001426 : 16, <class 'list'>
1001427 : 50, <class 'list'>
1001428 : 3, <class 'list'>
1001429 : 29, <class 'list'>
1001430 : 79, <class 'list'>
1001431 : 12, <class 'list'>
1001432 : 10, <class 'list'>
1001433 : 9, <class 'list'>
1001434 : 3, <class 'list'>
1001435 : 20, <class 'list'>
1001436 : 22, <class 'list'>
1001437 : 9, <class 'list'>
1001438 : 19, <class 'list'>
1001439 : 14, <class 'list'>
1001440 : 31, <class 'list'>
1001441 : 32, <class

In [149]:
df_2020 = service_info_2020[service_info_2020['상권_코드'] == 1000013]
df_2018 = service_info_2018[service_info_2018['상권_코드'] == 1000013]

n1 = len(df_2020['서비스_업종_코드'].unique())
n2 = len(df_2018['서비스_업종_코드'].unique())

print(n1,n2)

29 29


In [142]:
# 2020
list_2020 = []
for m in mlist:
    df = service_info_2020[service_info_2020['상권_코드'] == m]
    for j in df.index:
        if df.loc[j,'서비스_업종_코드'] in globals()['market_{}'.format(m)]:
            continue
        else:
            list_2020.append(j)

service_info_2020 = service_info_2020.drop(list_2020).reset_index(drop = True)

In [143]:
# 2019
list_2019 = []
for m in mlist:
    df = service_info_2019[service_info_2019['상권_코드'] == m]
    for j in df.index:
        if df.loc[j,'서비스_업종_코드'] in globals()['market_{}'.format(m)]:
            continue
        else:
            list_2019.append(j)

service_info_2019 = service_info_2019.drop(list_2019).reset_index(drop = True)

In [144]:
# 2018
list_2018 = []
for m in mlist:
    df = service_info_2018[service_info_2018['상권_코드'] == m]
    for j in df.index:
        if df.loc[j,'서비스_업종_코드'] in globals()['market_{}'.format(m)]:
            continue
        else:
            list_2018.append(j)

service_info_2018 = service_info_2018.drop(list_2018).reset_index(drop = True)

In [145]:
# 2017
list_2017 = []
for m in mlist:
    df = service_info_2017[service_info_2017['상권_코드'] == m]
    for j in df.index:
        if df.loc[j,'서비스_업종_코드'] in globals()['market_{}'.format(m)]:
            continue
        else:
            list_2017.append(j)

service_info_2017 = service_info_2017.drop(list_2017).reset_index(drop = True)

In [146]:
# 2016
list_2016 = []
for m in mlist:
    df = service_info_2016[service_info_2016['상권_코드'] == m]
    for j in df.index:
        if df.loc[j,'서비스_업종_코드'] in globals()['market_{}'.format(m)]:
            continue
        else:
            list_2016.append(j)
            
service_info_2016 = service_info_2016.drop(list_2016).reset_index(drop = True)        

In [147]:
# 2015
list_2015 = []
for m in mlist:
    df = service_info_2015[service_info_2015['상권_코드'] == m]
    for j in df.index:
        if df.loc[j,'서비스_업종_코드'] in globals()['market_{}'.format(m)]:
            continue
        else:
            list_2015.append(j)
            
service_info_2015 = service_info_2015.drop(list_2015).reset_index(drop = True)

In [148]:
df_2020 = service_info_2020[service_info_2020['상권_코드'] == 1000013]
df_2018 = service_info_2016[service_info_2016['상권_코드'] == 1000013]

n1 = len(df_2020['서비스_업종_코드'].unique())
n2 = len(df_2018['서비스_업종_코드'].unique())

print(n1,n2)

29 29


### (2) 업종별 추정매출 데이터

In [151]:
# 2020년 = service_sales_2020, 
# 2019년 = service_sales_2019, 
# .... 
# 2015년 = service_sales_2015

for i in range(2015, 2021):
    globals()['service_sales_{}'.format(i)] = pd.read_csv(path + str(i) + '.csv', encoding = 'cp949')

In [152]:
# 확인

service_sales_2020

기준_년_코드  기준_분기_코드 상권_구분_코드 상권_구분_코드_명    상권_코드      상권_코드_명 서비스_업종_코드  \
0         2020         2        U       관광특구  1001496  강남 마이스 관광특구  CS300043   
1         2020         2        U       관광특구  1001496  강남 마이스 관광특구  CS300032   
2         2020         2        U       관광특구  1001496  강남 마이스 관광특구  CS300031   
3         2020         2        U       관광특구  1001496  강남 마이스 관광특구  CS300028   
4         2020         2        U       관광특구  1001496  강남 마이스 관광특구  CS300027   
...        ...       ...      ...        ...      ...          ...       ...   
66506     2020         1        A       골목상권  1000001          계동길  CS100005   
66507     2020         1        A       골목상권  1000001          계동길  CS100004   
66508     2020         1        A       골목상권  1000001          계동길  CS100003   
66509     2020         1        A       골목상권  1000001          계동길  CS100002   
66510     2020         1        A       골목상권  1000001          계동길  CS100001   

      서비스_업종_코드_명     당월_매출_금액  당월_매출_건수  ...  시간대_건수~24_매출_건수  남성_매출_건수  \
0          전자상거래업      6415600       169  ...                0        63   
1            가전제품   1853749205      8319  ...              132      4331   
2              가구  10400909742      5270  ...               35      2458   
3              화초    807257363     28506  ...              580      9007   
4            섬유제품   1410497003     41063  ...             1445      8135   
...           ...          ...       ...  ...              ...       ...   
66506         제과점    257814129     27092  ...             1205     10426   
66507       양식음식점    156728306      3589  ...               33      1281   
66508       일식음식점     55484866      1272  ...              241       622   
66509       중식음식점     25767161      1413  ...                0       592   
66510       한식음식점    431442455     18548  ...              345      8752   

       여성_매출_건수  연령대_10_매출_건수  연령대_20_매출_건수  연령대_30_매출_건수  연령대_40_매출_건수  \
0           106             0            10           139            10   
1          3363            66          2146          2181          1234   
2          2812             0             0           245          1441   
3         18931           232         10926         10020          3876   
4         32928          1478         20434          8248          5427   
...         ...           ...           ...           ...           ...   
66506     14742           234          4316          7089          6188   
66507      2014             5          1361          1116           331   
66508       508             0           134           245           275   
66509       694           145           368           240           368   
66510      7776           611          3828          4356          3428   

       연령대_50_매출_건수  연령대_60_이상_매출_건수  점포수  
0                 5                5    8  
1              1062             1003    8  
2              1722             1862    5  
3              2101              781    6  
4              4241             1235    6  
...             ...              ...  ...  
66506          4351             2989    4  
66507           325              157    3  
66508           314              162    3  
66509            96               69    1  
66510          2614             1690   19  

[66511 rows x 80 columns]

In [153]:
# 분기 분리

for i in range(2015, 2021):
    df = globals()['service_sales_{}'.format(i)]
    for j in range(1, 5):
        df_2 = df[df['기준_분기_코드'] == j]
        n1 = len(df_2['상권_코드'].unique())
        print("{}_{} : {}".format(i, j, n1))
        if i == 2020 and j == 2:
            break
        
        

2015_1 : 1491
2015_2 : 1491
2015_3 : 1491
2015_4 : 1491
2016_1 : 1492
2016_2 : 1493
2016_3 : 1493
2016_4 : 1491
2017_1 : 1492
2017_2 : 1492
2017_3 : 1491
2017_4 : 1491
2018_1 : 1490
2018_2 : 1490
2018_3 : 1491
2018_4 : 1491
2019_1 : 1487
2019_2 : 1485
2019_3 : 1486
2019_4 : 1487
2020_1 : 1485
2020_2 : 1485


In [154]:
# 2015 ~ 2020 추정매출, 폐업률 수직 통합

total_service_info = pd.DataFrame()
total_service_sales = pd.DataFrame()

# 점포 데이터 통합 (2015 ~ 2020)
for i in range (2015, 2021):
    total_service_info = total_service_info.append(globals()['service_info_{}'.format(i)], ignore_index = True)

# 불필요한 column 제거
total_service_info = total_service_info.drop(['상권_구분_코드', '상권_구분_코드_명', '서비스_업종_코드_명'], axis=1)

# 추정매출 데이터 통합 (2015 ~ 2020)
for i in range (2015, 2021):
    total_service_sales = total_service_sales.append(globals()['service_sales_{}'.format(i)], ignore_index = True)

# 불필요한 column 제거
total_service_sales = total_service_sales.drop(['상권_구분_코드', '상권_구분_코드_명', '서비스_업종_코드_명'], axis=1)

In [155]:
# 결과 확인
# 통합 점포 데이터

total_service_info

기준_년_코드  기준_분기_코드    상권_코드       상권_코드_명 서비스_업종_코드  점포_수  유사_업종_점포_수  \
0           2015         1  1001490       길동복조리시장  CS300011    33          33   
1           2015         1  1001046  서울 관악구 신림역_4  CS300001    18          18   
2           2015         1  1000609        개봉로11길  CS100001     9           9   
3           2015         1  1000910       도산대로81길  CS100004     9           9   
4           2015         1  1000862        방배로18길  CS100002     1           2   
...          ...       ...      ...           ...       ...   ...         ...   
1259011     2020         1  1000001           계동길  CS100005     2           4   
1259012     2020         1  1000001           계동길  CS100004     3           3   
1259013     2020         1  1000001           계동길  CS100003     3           3   
1259014     2020         1  1000001           계동길  CS100002     0           1   
1259015     2020         1  1000001           계동길  CS100001    19          19   

         개업_율  개업_점포_수  폐업_률  폐업_점포_수  프랜차이즈_점포_수  
0           0        0     3        1           0  
1           0        0     0        0           0  
2           0        0     0        0           0  
3          22        2    33        3           0  
4           0        0     0        0           1  
...       ...      ...   ...      ...         ...  
1259011     0        0     0        0           2  
1259012     0        0     0        0           0  
1259013     0        0     0        0           0  
1259014     0        0     0        0           1  
1259015     0        0     5        1           0  

[1259016 rows x 12 columns]

In [156]:
total_service_info.isnull().sum()

기준_년_코드       0
기준_분기_코드      0
상권_코드         0
상권_코드_명       0
서비스_업종_코드     0
점포_수          0
유사_업종_점포_수    0
개업_율          0
개업_점포_수       0
폐업_률          0
폐업_점포_수       0
프랜차이즈_점포_수    0
dtype: int64

In [157]:
# 결과 확인
# 통합 추정매출 데이터

total_service_sales

기준_년_코드  기준_분기_코드    상권_코드  상권_코드_명 서비스_업종_코드    당월_매출_금액  당월_매출_건수  \
0          2015         1  1001018  포이사거리_2  CS300021  3390014653     88106   
1          2015         1  1000481  성미산로11길  CS300011   477810637      1339   
2          2015         1  1000440   세검정로1길  CS200008    29303647       582   
3          2015         1  1001458     인헌시장  CS100005   312469941     38207   
4          2015         1  1000756    사당로8길  CS200016    38474191      2583   
...         ...       ...      ...      ...       ...         ...       ...   
862311     2020         1  1000001      계동길  CS100005   257814129     27092   
862312     2020         1  1000001      계동길  CS100004   156728306      3589   
862313     2020         1  1000001      계동길  CS100003    55484866      1272   
862314     2020         1  1000001      계동길  CS100002    25767161      1413   
862315     2020         1  1000001      계동길  CS100001   431442455     18548   

        주중_매출_비율  주말_매출_비율  월요일_매출_비율  ...  시간대_건수~24_매출_건수  남성_매출_건수  \
0             98         3         22  ...               41     26390   
1             86        14         23  ...                9       781   
2             85        15         17  ...                0       258   
3             68        32         11  ...             3475     13485   
4             61        39         11  ...              854      2317   
...          ...       ...        ...  ...              ...       ...   
862311        76        24         16  ...             1205     10426   
862312        42        58          6  ...               33      1281   
862313        83        17         14  ...              241       622   
862314        78        22         12  ...                0       592   
862315        79        21         13  ...              345      8752   

        여성_매출_건수  연령대_10_매출_건수  연령대_20_매출_건수  연령대_30_매출_건수  연령대_40_매출_건수  \
0          18879           385          8047         13300         11808   
1            406             0            64           472           216   
2            324             0             0           189            18   
3          22883           352          6467         13345          9907   
4            197           165           827           664           544   
...          ...           ...           ...           ...           ...   
862311     14742           234          4316          7089          6188   
862312      2014             5          1361          1116           331   
862313       508             0           134           245           275   
862314       694           145           368           240           368   
862315      7776           611          3828          4356          3428   

        연령대_50_매출_건수  연령대_60_이상_매출_건수  점포수  
0               8195             3532   19  
1                328              108    7  
2                197              178    1  
3               4002             2294    3  
4                237               77    1  
...              ...              ...  ...  
862311          4351             2989    4  
862312           325              157    3  
862313           314              162    3  
862314            96               69    1  
862315          2614             1690   19  

[862316 rows x 77 columns]

In [158]:
# 공통되는 column 찾기

c_info = set(total_service_info.columns)
c_sales = set(total_service_sales.columns)
c_same = list(set.intersection(c_info, c_sales))

c_same

['상권_코드', '기준_분기_코드', '상권_코드_명', '서비스_업종_코드', '기준_년_코드']

In [159]:
# 통합 점포 데이터 + 통합 추정매출 데이터 = total_df

total_df = pd.merge(total_service_info, total_service_sales, how = 'left', on = c_same)

In [160]:
# 결과 확인
# 전체 통합 데이터

total_df

기준_년_코드  기준_분기_코드    상권_코드       상권_코드_명 서비스_업종_코드  점포_수  유사_업종_점포_수  \
0           2015         1  1001490       길동복조리시장  CS300011    33          33   
1           2015         1  1001046  서울 관악구 신림역_4  CS300001    18          18   
2           2015         1  1000609        개봉로11길  CS100001     9           9   
3           2015         1  1000910       도산대로81길  CS100004     9           9   
4           2015         1  1000862        방배로18길  CS100002     1           2   
...          ...       ...      ...           ...       ...   ...         ...   
1259011     2020         1  1000001           계동길  CS100005     2           4   
1259012     2020         1  1000001           계동길  CS100004     3           3   
1259013     2020         1  1000001           계동길  CS100003     3           3   
1259014     2020         1  1000001           계동길  CS100002     0           1   
1259015     2020         1  1000001           계동길  CS100001    19          19   

         개업_율  개업_점포_수  폐업_률  ...  시간대_건수~24_매출_건수  남성_매출_건수  여성_매출_건수  \
0           0        0     3  ...            643.0    4389.0   11072.0   
1           0        0     0  ...          48014.0   79772.0  108601.0   
2           0        0     0  ...            664.0    1620.0     951.0   
3          22        2    33  ...           1908.0    4735.0    6426.0   
4           0        0     0  ...             75.0    1726.0     892.0   
...       ...      ...   ...  ...              ...       ...       ...   
1259011     0        0     0  ...           1205.0   10426.0   14742.0   
1259012     0        0     0  ...             33.0    1281.0    2014.0   
1259013     0        0     0  ...            241.0     622.0     508.0   
1259014     0        0     0  ...              0.0     592.0     694.0   
1259015     0        0     5  ...            345.0    8752.0    7776.0   

         연령대_10_매출_건수  연령대_20_매출_건수  연령대_30_매출_건수  연령대_40_매출_건수  연령대_50_매출_건수  \
0                97.0        1205.0        2267.0        4759.0        4584.0   
1              1293.0       68332.0       55144.0       26564.0       23568.0   
2                 7.0         199.0         509.0         785.0         662.0   
3                81.0        3347.0        4750.0        1784.0         929.0   
4                16.0         509.0        1021.0         731.0         220.0   
...               ...           ...           ...           ...           ...   
1259011         234.0        4316.0        7089.0        6188.0        4351.0   
1259012           5.0        1361.0        1116.0         331.0         325.0   
1259013           0.0         134.0         245.0         275.0         314.0   
1259014         145.0         368.0         240.0         368.0          96.0   
1259015         611.0        3828.0        4356.0        3428.0        2614.0   

         연령대_60_이상_매출_건수   점포수  
0                 2549.0  33.0  
1                13473.0  18.0  
2                  407.0   9.0  
3                  276.0  10.0  
4                  121.0   2.0  
...                  ...   ...  
1259011           2989.0   4.0  
1259012            157.0   3.0  
1259013            162.0   3.0  
1259014             69.0   1.0  
1259015           1690.0  19.0  

[1259016 rows x 84 columns]

## 3. 결측치 확인
### (1) 각 연도, 분기 데이터에 1496개 상권이 모두 포함되어 있는지 확인

In [163]:
# 연도 분리
for i in range(2015, 2021):
    globals()['df_{}'.format(i)] = total_df[total_df['기준_년_코드'] == i]
    
# 분기 분리
for i in range(2015, 2021):
    for j in range(1, 5):
        globals()['df_{}_{}'.format(i, j)] = globals()['df_{}'.format(i)][globals()['df_{}'.format(i)]['기준_분기_코드'] == j]
        if i == 2020 and j == 2:
            break
            

In [164]:
# 연도별, 분기별 상권 개수 출력

df_list = [df_2020_1, df_2020_2, df_2019_1, df_2019_2, df_2019_3, df_2019_4, df_2018_1, df_2018_2, df_2018_3, df_2018_4, df_2017_1, 
        df_2017_2, df_2017_3, df_2017_4, df_2016_1, df_2016_2, df_2016_3, df_2016_4, df_2015_1, df_2015_2, df_2015_3, df_2015_4]

for df in df_list:
    print(len(df['상권_코드'].unique()))

1496
1496
1496
1496
1496
1496
1496
1496
1496
1496
1496
1496
1496
1496
1496
1496
1496
1496
1496
1496
1496
1496


## (2) column별 결측치 확인

In [165]:
# 모든 column name

total_df.columns

Index(['기준_년_코드', '기준_분기_코드', '상권_코드', '상권_코드_명', '서비스_업종_코드', '점포_수',
       '유사_업종_점포_수', '개업_율', '개업_점포_수', '폐업_률', '폐업_점포_수', '프랜차이즈_점포_수',
       '당월_매출_금액', '당월_매출_건수', '주중_매출_비율', '주말_매출_비율', '월요일_매출_비율',
       '화요일_매출_비율', '수요일_매출_비율', '목요일_매출_비율', '금요일_매출_비율', '토요일_매출_비율',
       '일요일_매출_비율', '시간대_00~06_매출_비율', '시간대_06~11_매출_비율', '시간대_11~14_매출_비율',
       '시간대_14~17_매출_비율', '시간대_17~21_매출_비율', '시간대_21~24_매출_비율', '남성_매출_비율',
       '여성_매출_비율', '연령대_10_매출_비율', '연령대_20_매출_비율', '연령대_30_매출_비율',
       '연령대_40_매출_비율', '연령대_50_매출_비율', '연령대_60_이상_매출_비율', '주중_매출_금액',
       '주말_매출_금액', '월요일_매출_금액', '화요일_매출_금액', '수요일_매출_금액', '목요일_매출_금액',
       '금요일_매출_금액', '토요일_매출_금액', '일요일_매출_금액', '시간대_00~06_매출_금액',
       '시간대_06~11_매출_금액', '시간대_11~14_매출_금액', '시간대_14~17_매출_금액',
       '시간대_17~21_매출_금액', '시간대_21~24_매출_금액', '남성_매출_금액', '여성_매출_금액',
       '연령대_10_매출_금액', '연령대_20_매출_금액', '연령대_30_매출_금액', '연령대_40_매출_금액',
       '연령대_50_매출_금액', '연령대_60_이상_매출_금액', '주중_매출_건수', '주말_매출_건수', '월요일_매출_건수',
       

In [166]:
# 모든 84개 열에 대한 각각의 결측치 수

for column in total_df.columns:
    n_of_null = total_df[column].isnull().sum()
    print("{} : {}".format(column, n_of_null))

기준_년_코드 : 0
기준_분기_코드 : 0
상권_코드 : 0
상권_코드_명 : 0
서비스_업종_코드 : 0
점포_수 : 0
유사_업종_점포_수 : 0
개업_율 : 0
개업_점포_수 : 0
폐업_률 : 0
폐업_점포_수 : 0
프랜차이즈_점포_수 : 0
당월_매출_금액 : 483875
당월_매출_건수 : 483875
주중_매출_비율 : 483875
주말_매출_비율 : 483875
월요일_매출_비율 : 483875
화요일_매출_비율 : 483875
수요일_매출_비율 : 483875
목요일_매출_비율 : 483875
금요일_매출_비율 : 483875
토요일_매출_비율 : 483875
일요일_매출_비율 : 483875
시간대_00~06_매출_비율 : 483875
시간대_06~11_매출_비율 : 483875
시간대_11~14_매출_비율 : 483875
시간대_14~17_매출_비율 : 483875
시간대_17~21_매출_비율 : 483875
시간대_21~24_매출_비율 : 483875
남성_매출_비율 : 483875
여성_매출_비율 : 483875
연령대_10_매출_비율 : 483875
연령대_20_매출_비율 : 483875
연령대_30_매출_비율 : 483875
연령대_40_매출_비율 : 483875
연령대_50_매출_비율 : 483875
연령대_60_이상_매출_비율 : 483875
주중_매출_금액 : 483875
주말_매출_금액 : 483875
월요일_매출_금액 : 483875
화요일_매출_금액 : 483875
수요일_매출_금액 : 483875
목요일_매출_금액 : 483875
금요일_매출_금액 : 483875
토요일_매출_금액 : 483875
일요일_매출_금액 : 483875
시간대_00~06_매출_금액 : 483875
시간대_06~11_매출_금액 : 483875
시간대_11~14_매출_금액 : 483875
시간대_14~17_매출_금액 : 483875
시간대_17~21_매출_금액 : 483875
시간대_21~24_매출_금액 : 483875
남성_매출_금액 : 48

In [167]:
total_df.shape

(1259016, 84)